### PR TITLE
Rename `pygame.typing.Coordinate` to `Point`, fix related grammar

### PR DESCRIPTION
**Pull request by aatle**
As per discussion on discord, the Coordinate class is grammatically incorrect and Point would be a better name for various reasons.

Rename `Coordinate` to `Point` and `IntCoordinate` to `IntPoint` in code. Mostly in `.pyi` files, and a few test modules.
(Non-python code was not checked.)

Fix grammar where "coordinate" was put when "point" or "coordinates" was intended, in pygame docs.

### CODE REVIEW

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for contributing :tada:

File: docs/reST/ref/gfxdraw.rst [hunk 1]
```diff
@@ -535,18 +535,18 @@ For example:
    Draws a filled polygon on the given surface. For an unfilled polygon use
    :meth:`polygon`.
 
-   The adjacent coordinates in the ``points`` argument, as well as the first
+   The adjacent pooints in the ``points`` argument, as well as the first
```
Comment (bilhox) [inline docs/reST/ref/gfxdraw.rst:538]: small typo

**Review by bilhox — CHANGES_REQUESTED** [review]
LGTM!

Just a small typo to fix 💀

File: docs/reST/ref/gfxdraw.rst (hunk 1 above)
Comment (aatle) [reply docs/reST/ref/gfxdraw.rst]: fixed

**Review by bilhox — APPROVED** [review]
LGTM

**Pull request merged**